test/test_fwd.py

```diff
@@ -83,6 +83,16 @@ def run_worker(args):
             f"equivalence test passed {torch.sum(out)} ref {torch.sum(ref_out)}"
         )
 
+    # Test qualname mapping
+    sd = stage.submod.state_dict()
+    print(f"Rank {args.rank} state dict keys: {sd.keys()}")
+    remapped_keys = [stage.remap_qualname(k) for k in sd.keys()]
+    print(f"Rank {args.rank} remapped keys: {remapped_keys}")
+    # Confirm remapped keys are consistent with original model
+    old_keys = mod.state_dict().keys()
+    assert all(rk in old_keys for rk in remapped_keys)
```

Comment:
nit: wouldn't this evaluate to be `True` if remapped_keys was a subset of old_keys.

Maybe a better check would be sorted(remapped_keys) == sorted(list(old_keys))

Reply:
`remapped_keys` is always a subset of `old_keys`, because `remapped_keys` are only keys of a stage, while `old_keys` contains keys of the full original model.